include/bluetooth/gatt.h

```diff
@@ -699,6 +699,29 @@ ssize_t bt_gatt_attr_read_cpf(struct bt_conn *conn,
 int bt_gatt_notify(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 		   const void *data, u16_t len);
 
+
+/** @brief Notification complete result callback.
+ *
+ *  @param conn Connection object.
+ */
+typedef void (*bt_gatt_notify_complete_func_t) (struct bt_conn *conn);
+
+/** @brief Notify attribute value change with callback.
+ *
+ *  This function works in the same way as @ref bt_gatt_notify.
+ *  With the addition that after sending the notification the
+ *  callback function will be called.
+ *
+ *  @param conn Connection object.
+ *  @param attr Characteristic or Characteristic Value attribute.
+ *  @param data Pointer to Attribute data.
+ *  @param len Attribute value length.
+ *  @param func Notification value callback.
+ */
+int bt_gatt_notify_with_cb(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+			   const void *data, u16_t len,
+			   bt_gatt_notify_complete_func_t func);
```

Comment:
If we are going to do this then why not have a _param struct, perhaps it doesn't make sense to have bt_att_req but otherwise it is pretty similar.

Reply:
You seem to have missed the recent discussion and reasoning about this

Reply:
@Vudentz note that unlike our other asynchronous callback "param" structs, the way it was used in the previous revision was different: the struct was not persistent (it didn't need to be), rather it was just used to pass function parameters.